server/src/main/java/org/opensearch/gateway/PrimaryShardAllocator.java

```diff
@@ -313,6 +318,10 @@ private static ShardStoreInfo shardStoreInfo(NodeGatewayStartedShards nodeShardS
         NodeGatewayStartedShards::primary
     ).reversed();
 
+    private static final Comparator<NodeGatewayStartedShards> HIGHEST_REPLICATION_FIRST_CHECKPOINT_COMPARATOR = Comparator.nullsLast(
```

Comment:
Need to change `HIGHEST_REPLICATION_FIRST_CHECKPOINT_COMPARATOR` -> `HIGHEST_REPLICATION_CHECKPOINT_FIRST_COMPARATOR`